In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
from utils import *
PATH = "/Users/danil/BSE/Tesis/dataSent12_21Good/"
companies_ranked = pd.read_csv(PATH + "CompaniesrankedMarketCap.csv")
big_sent_all = pd.read_csv(PATH + "USbig_Sent12_21.csv")
big_sent_all = pd.DataFrame(big_sent_all)[["date", "stock", "linscr"]]

In [2]:
big_sent_all

date stock   linscr
0       2012-01-04T00:00:00Z   AAL  75.0419
1       2012-01-05T00:00:00Z   AAL  53.3333
2       2012-01-06T00:00:00Z   AAL  74.3750
3       2012-01-07T00:00:00Z   AAL  70.9177
4       2012-01-08T00:00:00Z   AAL   0.0000
...                      ...   ...      ...
289496  2021-11-27T00:00:00Z    ET   0.0000
289497  2021-11-28T00:00:00Z    ET   0.0000
289498  2021-11-29T00:00:00Z    ET   0.0000
289499  2021-11-30T00:00:00Z    ET   0.0000
289500  2021-12-01T00:00:00Z    ET   0.0000

[289501 rows x 3 columns]

In [3]:
big_sent_pivot = big_sent_all.pivot(index="date", columns="stock", values= "linscr")
big_sent_pivot

stock                     AAL     AAPL     ABBV      ABC      ABT      ADP  \
date                                                                         
2012-01-02T00:00:00Z      NaN      NaN      NaN      NaN      NaN      NaN   
2012-01-03T00:00:00Z      NaN  15.0997      NaN  83.3333  17.1028      NaN   
2012-01-04T00:00:00Z  75.0419  56.3951      NaN  66.7830  17.3095  47.9782   
2012-01-05T00:00:00Z  53.3333  60.5574      NaN  43.1752  52.4642   8.6337   
2012-01-06T00:00:00Z  74.3750  58.7690      NaN  90.0332  50.0000  39.9165   
...                       ...      ...      ...      ...      ...      ...   
2021-11-27T00:00:00Z  20.3302  54.3086  75.7296  51.7157  64.4993   0.0000   
2021-11-28T00:00:00Z  38.0634  55.4475      NaN      NaN      NaN      NaN   
2021-11-29T00:00:00Z  44.3530      NaN      NaN  35.9823      NaN      NaN   
2021-11-30T00:00:00Z  45.4783  67.7697  45.3517  40.8998  54.7860  33.3333   
2021-12-01T00:00:00Z  18.1411  61.0514  30.2032  47.0266  59.5313  42.6829   

stock                     AIG       AMD     AMZN      AXP  ...      UAL  \
date                                                       ...            
2012-01-02T00:00:00Z      NaN       NaN      NaN      NaN  ...  10.9553   
2012-01-03T00:00:00Z      NaN       NaN      NaN      NaN  ...  35.7657   
2012-01-04T00:00:00Z  18.0340   57.2778  64.4571  77.0397  ...  47.1125   
2012-01-05T00:00:00Z  88.0108   60.1085  19.9874  65.6233  ...  43.1721   
2012-01-06T00:00:00Z  69.7269  100.0000  52.5774  30.1755  ...  28.7344   
...                       ...       ...      ...      ...  ...      ...   
2021-11-27T00:00:00Z  58.3012   73.7678  30.2219  10.0723  ...  15.6070   
2021-11-28T00:00:00Z      NaN   83.5694  54.2548      NaN  ...  16.4894   
2021-11-29T00:00:00Z      NaN  100.0000  14.0921  20.0000  ...  30.7571   
2021-11-30T00:00:00Z  33.2589   85.7355  31.9313  55.3991  ...  26.4182   
2021-12-01T00:00:00Z  81.7725   73.6452  57.4477   0.0000  ...  16.4318   

stock                      UNH      UPS       USB         V       VZ      WFC  \
date                                                                            
2012-01-02T00:00:00Z       NaN      NaN       NaN       NaN      NaN      NaN   
2012-01-03T00:00:00Z       NaN      NaN  100.0000       NaN  14.2857  53.8188   
2012-01-04T00:00:00Z    0.0000  66.7323   61.1930  100.0000  41.5813   0.0000   
2012-01-05T00:00:00Z    0.0000  78.4615   89.2930    0.0000  51.3410  51.3178   
2012-01-06T00:00:00Z   68.7500   0.0000   76.0074    0.0000  36.2199  17.2306   
...                        ...      ...       ...       ...      ...      ...   
2021-11-27T00:00:00Z       NaN  19.9309   42.9630       NaN   0.0000  20.4201   
2021-11-28T00:00:00Z  100.0000   8.3744       NaN       NaN  33.3500   0.0000   
2021-11-29T00:00:00Z       NaN   8.3744  100.0000       NaN      NaN   0.0000   
2021-11-30T00:00:00Z   84.7019  55.1502   55.9799   55.2279  72.8938  76.6631   
2021-12-01T00:00:00Z   29.7989  30.2960   68.5516    0.0000  58.0229  55.5006   

stock                     WMT       WY      XOM  
date                                             
2012-01-02T00:00:00Z      NaN      NaN  12.2807  
2012-01-03T00:00:00Z  14.2857      NaN   2.2519  
2012-01-04T00:00:00Z  21.4437   0.0000  14.3610  
2012-01-05T00:00:00Z  28.2662  69.4719  14.4794  
2012-01-06T00:00:00Z  39.2602      NaN  21.1067  
...                       ...      ...      ...  
2021-11-27T00:00:00Z  54.6990   0.0000   0.0000  
2021-11-28T00:00:00Z  65.2119      NaN      NaN  
2021-11-29T00:00:00Z  39.7059      NaN      NaN  
2021-11-30T00:00:00Z  42.1815  50.0000  39.1154  
2021-12-01T00:00:00Z  18.2261  55.1066  16.8894  

[3620 rows x 108 columns]

We check viable stocks with less than 20% missing values. We note that there aren't many (28) from the original 108. Next we want to know if we remove some consecutive early years, we get a lower proportion of NaN's.

In [4]:
get_viable_stocks(big_sent_pivot, 0.2)

array(['AAPL', 'ABC', 'AMZN', 'BAC', 'C', 'COO', 'DIS', 'ET', 'F', 'FB',
       'GE', 'GM', 'GOOG', 'GS', 'HD', 'IBM', 'ICE', 'INTC', 'JPM', 'MS',
       'MSCI', 'MSFT', 'NFLX', 'T', 'TSLA', 'VZ', 'WFC', 'WMT'],
      dtype=object)

In [5]:
big_year = get_year(big_sent_pivot)

In [7]:
big_year

stock                          AAL     AAPL     ABBV      ABC      ABT  \
date                                                                     
2012-01-02 00:00:00+00:00      NaN      NaN      NaN      NaN      NaN   
2012-01-03 00:00:00+00:00      NaN  15.0997      NaN  83.3333  17.1028   
2012-01-04 00:00:00+00:00  75.0419  56.3951      NaN  66.7830  17.3095   
2012-01-05 00:00:00+00:00  53.3333  60.5574      NaN  43.1752  52.4642   
2012-01-06 00:00:00+00:00  74.3750  58.7690      NaN  90.0332  50.0000   
...                            ...      ...      ...      ...      ...   
2021-11-27 00:00:00+00:00  20.3302  54.3086  75.7296  51.7157  64.4993   
2021-11-28 00:00:00+00:00  38.0634  55.4475      NaN      NaN      NaN   
2021-11-29 00:00:00+00:00  44.3530      NaN      NaN  35.9823      NaN   
2021-11-30 00:00:00+00:00  45.4783  67.7697  45.3517  40.8998  54.7860   
2021-12-01 00:00:00+00:00  18.1411  61.0514  30.2032  47.0266  59.5313   

stock                          ADP      AIG       AMD     AMZN      AXP  ...  \
date                                                                     ...   
2012-01-02 00:00:00+00:00      NaN      NaN       NaN      NaN      NaN  ...   
2012-01-03 00:00:00+00:00      NaN      NaN       NaN      NaN      NaN  ...   
2012-01-04 00:00:00+00:00  47.9782  18.0340   57.2778  64.4571  77.0397  ...   
2012-01-05 00:00:00+00:00   8.6337  88.0108   60.1085  19.9874  65.6233  ...   
2012-01-06 00:00:00+00:00  39.9165  69.7269  100.0000  52.5774  30.1755  ...   
...                            ...      ...       ...      ...      ...  ...   
2021-11-27 00:00:00+00:00   0.0000  58.3012   73.7678  30.2219  10.0723  ...   
2021-11-28 00:00:00+00:00      NaN      NaN   83.5694  54.2548      NaN  ...   
2021-11-29 00:00:00+00:00      NaN      NaN  100.0000  14.0921  20.0000  ...   
2021-11-30 00:00:00+00:00  33.3333  33.2589   85.7355  31.9313  55.3991  ...   
2021-12-01 00:00:00+00:00  42.6829  81.7725   73.6452  57.4477   0.0000  ...   

stock                           UNH      UPS       USB         V       VZ  \
date                                                                        
2012-01-02 00:00:00+00:00       NaN      NaN       NaN       NaN      NaN   
2012-01-03 00:00:00+00:00       NaN      NaN  100.0000       NaN  14.2857   
2012-01-04 00:00:00+00:00    0.0000  66.7323   61.1930  100.0000  41.5813   
2012-01-05 00:00:00+00:00    0.0000  78.4615   89.2930    0.0000  51.3410   
2012-01-06 00:00:00+00:00   68.7500   0.0000   76.0074    0.0000  36.2199   
...                             ...      ...       ...       ...      ...   
2021-11-27 00:00:00+00:00       NaN  19.9309   42.9630       NaN   0.0000   
2021-11-28 00:00:00+00:00  100.0000   8.3744       NaN       NaN  33.3500   
2021-11-29 00:00:00+00:00       NaN   8.3744  100.0000       NaN      NaN   
2021-11-30 00:00:00+00:00   84.7019  55.1502   55.9799   55.2279  72.8938   
2021-12-01 00:00:00+00:00   29.7989  30.2960   68.5516    0.0000  58.0229   

stock                          WFC      WMT       WY      XOM  Year  
date                                                                 
2012-01-02 00:00:00+00:00      NaN      NaN      NaN  12.2807  2012  
2012-01-03 00:00:00+00:00  53.8188  14.2857      NaN   2.2519  2012  
2012-01-04 00:00:00+00:00   0.0000  21.4437   0.0000  14.3610  2012  
2012-01-05 00:00:00+00:00  51.3178  28.2662  69.4719  14.4794  2012  
2012-01-06 00:00:00+00:00  17.2306  39.2602      NaN  21.1067  2012  
...                            ...      ...      ...      ...   ...  
2021-11-27 00:00:00+00:00  20.4201  54.6990   0.0000   0.0000  2021  
2021-11-28 00:00:00+00:00   0.0000  65.2119      NaN      NaN  2021  
2021-11-29 00:00:00+00:00   0.0000  39.7059      NaN      NaN  2021  
2021-11-30 00:00:00+00:00  76.6631  42.1815  50.0000  39.1154  2021  
2021-12-01 00:00:00+00:00  55.5006  18.2261  55.1066  16.8894  2021  

[3620 rows x 109 columns]